hiv-benchmarking/README.md

```diff
@@ -0,0 +1,86 @@
+- Using the `hsa_uniprot.list` file from [LinkDB](https://www.genome.jp/linkdb/) (this was done by downloading the link information between HSA and Uniprot), the HSA IDs are mapped to all the UniProt IDs corresponding to the HSA IDs. Note that this produced a 1-to-many mapping for some proteins
+- To remove the 1-to-many mapping and produce a 1-to-1 mapping instead, API calls to [genome.jp](genome.jp) are done again to filter out UniProt IDs that *don't* have SwissProt IDs i.e. protein IDs that haven't been 'manually reviewed'.
+- All these mappings are saved locally as csvs.
+
+
+#### 7. hiv05_comparison_ratios
+This notebook contains code to create comparison ratios for SPRAS ensemble pathways vs. the original publication pathway *and* SPRAS ensemble pathways vs. the KEGG pathway 
+- This is done by intersecting the nodes found in both pathways, and then comparing the number of nodes found in both to the number of nodes found in the individual original pathways
+
+
+#### 8. ensemble_node_maxfreq.ipynb
+This notebook contains code that loads the ensemble pathway and assigns the highest frequency associated with each node to the respective node.
+- The original SPRAS output ensemble pathway file assigns frequencies to edges. 
+- The result produced by this notebook assigns the maximum edge frequency associated with each node to the node, creating a max node frequency list that is saved as a CSV.
+
+
+#### 9. build_prc_kegg_hiv05.ipynb
+This notebook produces a precision recall curve using `scikitlearn`. This is done through the following steps:
+- Building the PRC uses the max node frequency produced from the ensemble pathway, and the KEGG to Swissprot/Uniprot mapping.
+- A vector of 0s and 1s is built that indicates whether the particular protein in the ensemble pathway is found in the KEGG pathway or not. This vector is called `y_kegg` and is attached to the ensemble pathway dataframe.
+    - The `max_freq` column in the dataframe is considered as the probabilities/scores and the `y_kegg` column in the dataframe is considered as the true label
+- Then, a precision recall curve is built.
+
+
+#### 10. build_prc_gene-ontology_hiv05_hiv060
+This notebook also produces several precision recall curves using `scikitlearn`. These are done from selecting 3 biological processes from the publication's list of observed biological processes and then taking the Uniprot list of all Gene Ontology proteins (for the human taxonomy) for that particular biological process.
+- A vector of 0s and 1s is built that indicates whether the particular protein in the ensemble pathway is found in the Gene Ontology list or not. This vector is called `y_go` and is attached to the ensemble pathway dataframe.
+    - The `max_freq` column in the dataframe is considered as the probabilities/scores and the `y_go` column in the dataframe is considered as the true label
+- Then, a precision recall curve is built.
+
+
+
```

Comment:
The data processing in the notebooks is clearly described above. It would also be helpful to have a new readme section for all the raw data files that provides the origin of those files. That could be grouped by data source when multiple files come from one GitHub repository or database.

Reply:
I have a README file in the hiv_raw_data folder that documents where each file comes from - I haven't grouped that by data source though

Reply:
Putting the sources in that readme works.